### ISSUE TITLE
.NET TimeSpan format does not work for milliseconds when precision is less than 3

### ISSUE THREAD

**Issue by FallingSnow** [moment/moment#3266]
```javascript
> time = moment.duration('00:00:15.72')
{ [Number: 15000]
  _milliseconds: 15000,
  _days: 0,
  _months: 0,
  _data: 
   { milliseconds: 0,
     seconds: 15,
     minutes: 0,
     hours: 0,
     days: 0,
     months: 0,
     years: 0 },
  _locale: 
   Locale {
     _ordinalParse: /\d{1,2}(th|st|nd|rd)/,
     ordinal: [Function],
     _abbr: 'en',
     _config: 
      { ordinalParse: /\d{1,2}(th|st|nd|rd)/,
        ordinal: [Function],
        abbr: 'en' },
     _ordinalParseLenient: /\d{1,2}(th|st|nd|rd)|\d{1,2}/ } }
```

only results in `15000`, not the correct `15720`, however

```javascript
> time = moment.duration('00:00:15.720')
{ [Number: 15720]
  _milliseconds: 15720,
  _days: 0,
  _months: 0,
  _data: 
   { milliseconds: 0,
     seconds: 15,
     minutes: 0,
     hours: 0,
     days: 0,
     months: 0,
     years: 0 },
  _locale: 
   Locale {
     _ordinalParse: /\d{1,2}(th|st|nd|rd)/,
     ordinal: [Function],
     _abbr: 'en',
     _config: 
      { ordinalParse: /\d{1,2}(th|st|nd|rd)/,
        ordinal: [Function],
        abbr: 'en' },
     _ordinalParseLenient: /\d{1,2}(th|st|nd|rd)|\d{1,2}/ } }
```

**Comment by icambron** [comment]
Yup, that does seem like it should work: https://msdn.microsoft.com/en-us/library/se73z7b9(v=vs.110).aspx. @mj1856 any thoughts on here? Obviously we can't fully conform to the 1-7 digits thing, but it seems reasonable that 1-3 work.

**Comment by maggiepint** [comment]
IMO anything from zero to seven digits should parse fine. If you get more than three, just ignore the rest, but it should parse.

Matt would be the best person to decide on this one though.

**Comment by mj1856** [comment]
Yes, we should support fewer than three decimals of precision for seconds.  Also, the fourth decimal, if present, should control rounding of the third.

Keep in mind that .NET's `TimeSpan.ToString` output limits hours to 23, and then provides 24 hour days which are also separated by a decimal - so we have to be careful only to work with the *last* decimal in the string.

```
moment.duration("06:00:00").asMilliseconds() // 21600000  (OK)
moment.duration("1.06:00:00").asMilliseconds() // 108000000  (OK)
moment.duration("1.06:00:00.1").asMilliseconds() // 108000000  (BAD)
moment.duration("1.06:00:00.12").asMilliseconds() //  108000000  (BAD)
moment.duration("1.06:00:00.123").asMilliseconds() // 108000123  (OK)
moment.duration("1.06:00:00.1239").asMilliseconds() // 108000123  (BAD)
```

**Comment by maggiepint** [comment]
closed in favor of #3276
